### PR TITLE
web: error modal when snapshot fails

### PR DESCRIPTION
**Pull request by jazzdan**
(Video is cropped a little strangely, sorry about that)

### CODE REVIEW

**Comment by victorwuky** [comment]
Thanks for creating this @jazzdan . This is much better than a silent error.

Can we make the error message more actionable for the user. Perhaps one of these:
- Ask them to copy and paste the error into a new Tilt issue so that we can investigate.
- Refer them to an existing Tilt issue to copy and paste the error. (Especially if we know what the error is, and we are planning to fix it in the future.)

Regarding the "unexpected character" error message. Can the user actually do anything with that information? If they can't, how about providing less information, such as "JSON.parse error". It seems that providing more detail that's not actionable would seem more frustrating to the user.

File: web/src/HUD.tsx [hunk 1]
```diff
@@ -190,9 +194,25 @@ class HUD extends Component<HudProps, HudState> {
...
+            if (err instanceof Error) {
+              this.setAppState({
+                showSnapshotModal: false,
+                error: `Error sending snapshot: ${err.message}`,
+              })
+            } else {
+              this.setAppState({
+                showSnapshotModal: false,
+                error:
+                  "Error sending snapshot. Check JavaScript console for more details",
```
Comment (victorwuky) [inline web/src/HUD.tsx:208]: @jazzdan : I don't think we should be asking a user to check the browser console for more details. If it's important for the user to see the details, can't we surface that information inline in the error here? I don't think we should be asking the user to debug the problem for us.

Comment (jazzdan) [reply web/src/HUD.tsx:208]: Yeah I'm not an expert in JS error handling but it seems like sometimes this error value isn't very helpful. Maybe I should do a `try`/`catch` too?

**Comment by jazzdan** [comment]
@victorwuky happy to add a message telling folks where to go for help, but I think being super smart about this (finding existing issues/parsing out the error and figuring out what kind of error it is) isn't worth it right now. I think that'd be a fair bit of work and shouldn't delay showing people the error.

**Comment by victorwuky** [comment]
> @victorwuky happy to add a message telling folks where to go for help, but I think being super smart about this (finding existing issues/parsing out the error and figuring out what kind of error it is) isn't worth it right now. I think that'd be a fair bit of work and shouldn't delay showing people the error.

Sounds good @jazzdan. I think directing folks to https://github.com/tilt-dev/tilt/issues or #tilt in http://slack.k8s.io/ would be great. (Or even https://tilt.dev/contact, though I think that may be removed since it's not in https://tilt.dev/index2.) Whatever you think makes more sense.

File: web/src/HUD.tsx (hunk 1 above)
Comment (victorwuky) [reply web/src/HUD.tsx:208]: @jazzdan : I was just trying to say I don't think we should show the words "Check JavaScript console for more details" to the user in their regular error flow. If that doesn't actually appear to the user, then it's fine. But if it does appear, I think we should remove it. 

I understand our users our technical and would certainly understand the message. But I don't think we should be exposing this in the main UX itself.

So simply:

```suggestion
                  "Error sending snapshot.",
```

Don't know what you mean by try/catch here. You'd have to ask someone else!

**Comment by jazzdan** [comment]
Now it looks like this:

Screenshot from 2020-06-03 14-01-49

File: web/src/ErrorModal.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            <a href="https://github.com/tilt-dev/tilt/issues/new">
+              opening a GitHub issue
+            </a>{" "}
+            or{" "}
+            <a href="https://docs.tilt.dev/debug_faq.html#where-can-i-ask-questions">
+              contacting us on Slack
+            </a>
+            .
+          </p>
+        </div>
```
Comment (victorwuky) [inline web/src/ErrorModal.tsx:51]: Copy and paste error @jazzdan ?

```suggestion
```

**Comment by victorwuky** [comment]
Looks perfect @jazzdan ! Thanks! (Except for that small dupe problem.)

**Comment by nicks** [comment]
re: the error message - ya, I think right now we expect the error message to always be non-actionable garbage. The problem is that the server does not report back errors in an intelligible way.

For now, can we just hard-code "Unknown error".

Then we can leave a TODO to decide on how the server should transmit error messages.

What I don't want to happen is that we write up a whole complicated decision tree that doesn't actually work, and then the next engineer who touches this code has to wonder why it doesn't work.

**Comment by jazzdan** [comment]
@nicks done

**Pull request merged**